validator/rpc/standard_api.go

```diff
@@ -39,6 +40,41 @@ func (s *Server) ListKeystores(
 	}, nil
 }
 
+// ImportKeystores allows for importing keystores into Prysm with their slashing protection history.
+func (s *Server) ImportKeystores(
+	ctx context.Context, req *ethpbservice.ImportKeystoresRequest,
+) (*ethpbservice.ImportKeystoresResponse, error) {
+	if !s.walletInitialized {
+		return nil, status.Error(codes.Internal, "Wallet not ready")
+	}
+	importer, ok := s.keymanager.(keymanager.Importer)
+	if !ok {
+		return nil, status.Error(codes.Internal, "Keymanager kind cannot import keys")
+	}
+	keystores := make([]*keymanager.Keystore, len(req.Keystores))
+	for i := 0; i < len(req.Keystores); i++ {
+		k := &keymanager.Keystore{}
+		if err := json.Unmarshal([]byte(req.Keystores[i]), k); err != nil {
+			return nil, status.Errorf(
+				codes.Internal, "Invalid keystore at index %d in request: %v", i, err,
+			)
+		}
+		keystores[i] = k
+	}
+	if req.SlashingProtection != "" {
+		if err := slashingprotection.ImportStandardProtectionJSON(
+			ctx, s.valDB, bytes.NewBuffer([]byte(req.SlashingProtection)),
+		); err != nil {
+			return nil, status.Errorf(codes.Internal, "Could not import slashing protection JSON: %v", err)
+		}
+	}
+	statuses, err := importer.ImportKeystores(ctx, keystores, req.Passwords)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not import keystores: %v", err)
+	}
+	return &ethpbservice.ImportKeystoresResponse{Statuses: statuses}, nil
```

Comment:
we can also include the message with the status, but I need to ask about including the public key here too in the response...

Reply:
Let's revisit in a future PR